ide/wg_ScriptView.ml

```diff
@@ -405,6 +407,72 @@ object (self)
       self#buffer#delete_mark (`MARK stop_mark)
     | _ -> ()
 
+  method apply_unicode_binding () =
+    let bindings = Preferences.get_unicode_bindings() in
+    (** Auxiliary function to test whether [s] is a prefix of [str];
+        Note that there might be overlap with wg_Completion::is_substring *)
+    let string_is_prefix s str =
+      let n = String.length s in
+      let m = String.length str in
+      if m < n
+        then false
+        else (s = String.sub str 0 n)
+      in
+    (* Auxiliary function to perform the lookup for a binding *)
+    let lookup prefix = (* lookup : string -> string option *)
+      let max_priority = 100000000 in
+      let cur_word = ref None in
+      let cur_prio = ref (max_priority+1) in
+      let test_binding (key, word, prio_opt) =
+        let prio =
+          match prio_opt with
+          | None -> max_priority
+          | Some p -> p
+          in
+        if string_is_prefix prefix key && prio < !cur_prio then begin
+          cur_word := Some word;
+          cur_prio := prio;
+        end in
+      List.iter test_binding bindings;
+      !cur_word
+      in
+    (** Auxiliary method to reach the beginning of line or the
+        nearest space before the iterator. *)
+    let rec get_line_start iter =
+      if iter#starts_line || Glib.Unichar.isspace iter#char then iter
+      else get_line_start iter#backward_char
+      in
+    (** Main action *)
+    let buffer = self#buffer in
+    let insert = buffer#get_iter `INSERT in
+    let insert_mark = buffer#create_mark ~left_gravity:false insert in
+    let () = buffer#begin_user_action () in
+    let word_to_insert =
+      try
+        let line_start = get_line_start insert in
+        let prev_backslash_search = insert#backward_search ~limit:line_start "\\" in
+        let backslash =
+          match prev_backslash_search with
+          | None -> raise Abort
+          | Some (backslash_start,backslash_stop) -> backslash_start
+          in
+        let prefix = backslash#get_text ~stop:insert in
+        let word =
+          match lookup prefix with
+          | None -> raise Abort
+          | Some word -> word
+          in
+        let was_deleted = buffer#delete_interactive ~start:backslash ~stop:insert () in
+        if not was_deleted then raise Abort;
+        word
+      with Abort -> " "
```

Comment:
There is a good reason to put a space. If you write normal text, between two sentences you would press DOT then SPACE then a Capitalized letter. In french, the dot is accessed using SHIFT. Thus, many people would keep the SHIFT key down for the 3 keystrokes (DOT, SPACE, First letter of next sentence). Hence the need to insert a space.

Besides, even for users with english keyboards, it is good to know that your shift+space was taken into account, but did not match any registered binding, rather than giving the feeling that the keystroke is just ignored. 

I'll had these comments in the source to make sure they don't get lost.